docs/guides/cloud/deploy/fly.rst

```diff
@@ -0,0 +1,34 @@
+.. _ref_guide_cloud_deploy_fly:
+
+======
+Fly.io
+======
+
+:edb-alt-title: Deploying applications built on EdgeDB Cloud to Fly.io
+
+1. Install the `Fly.io CLI <https://fly.io/docs/hands-on/install-flyctl/>`_
+2. Log in to Fly.io with ``flyctl auth login``
+3. Run ``flyctl launch`` to create a new app on Fly.io and configure it.
+   It will ask you to select a region and a name for your app. When done it will 
+   create a ``fly.toml`` file and a ``Dockerfile`` in your project directory.
+4. Open the ``Dockefile`` and add modify the ``build`` step to mount the EdgeDB 
+   Cloud secret key and instance name as environment variables:
+
+   .. code-block:: dockerfile-diff
+    :caption: Dockerfile
+  
+      # Build application
+    -  RUN pnpm run build
+    +  RUN --mount=type=secret,id=EDGEDB_INSTANCE \
+    +      --mount=type=secret,id=EDGEDB_SECRET_KEY \
+    +      EDGEDB_INSTANCE="$(cat /run/secrets/EDGEDB_INSTANCE)" \
+    +      EDGEDB_SECRET_KEY="$(cat /run/secrets/EDGEDB_SECRET_KEY)" \
+    +      pnpm run build
+
+5. Run ``flyctl deploy`` to deploy your app to Fly.io. You need to pass the 
+   secrets to the deployment command:
+
+   .. code-block:: bash
+
+    $ flyctl deploy --build-secret EDGEDB_INSTANCE="<EDGEDB_INSTANCE>" \
+        --build-secret EDGEDB_SECRET_KEY="<EDGEDB_SECRET_KEY>"
```

Comment:
The build secrets didn't work for me. It may be something particular to my app I'm deploying, but I don't think it is. I believe for any of our clients, the env vars will need to be available at runtime. This required me to set them with `flyctl secrets set EDGEDB_INSTANCE=val EDGEDB_SECRET_KEY=val` instead of by using `--build-secret` on the `deploy` command.

Even if this isn't a universal requirement, we should probably mention both and describe the use case for each. I'm not sure of a case where `--build-secret` will work, but I could be wrong on that.